src/main/java/com/github/feroult/gapi/spreadsheet/SpreadsheetAPIImpl.java

```diff
@@ -249,11 +252,21 @@ private void adjustWorksheetDimensions(SpreadsheetBatch updateSet, BatchOptions.
 					if (objects.size() - 1 >= j) {
 						value = objects.get(j).toString();
 					}
-					row.put(header.get(j).toString(), value);
+					row.put(header.get(j), value);
 				}
 				ret.add(row);
 			}
 		}
 		return ret;
 	}
+
+	private List<String> convertObjectToString(List<Object> objects) {
+		List<String> strings = new ArrayList<>();
+		objects.forEach(obj -> strings.add(normalize(obj.toString())));
+		return strings;
+	}
+
+	private String normalize(String str) {
+		return str.replace(" ", "").toLowerCase();
```

Comment:
we had problems before with this "auto normalization" and special characters that sometimes got removed (eg parenthesis, dashs, slashes).

Reply:
take a look at a function called normalize on the wavez project

Reply:
I saw the method, but it's a little different. The gapi's old version don't remove accents, just some special characters. I noticed that some characters was missing in my `normalize` function, so I did a refactoring.